addons/delivery/views/delivery_view.xml

```diff
@@ -97,12 +97,13 @@
                             <group name="delivery_details">
                                 <field name="product_id" context="{'default_detailed_type': 'service', 'default_sale_ok': False, 'default_purchase_ok': False, 'default_invoice_policy': 'order'}" />
                                 <field name="invoice_policy" widget="radio" attrs="{'invisible': ['|', ('delivery_type', 'in', ('fixed', 'base_on_rule')), ('integration_level', '=', 'rate')]}"/>
-                                <label for="margin" string="Margin on Rate"/>
-                                <div>
-                                    <field name="margin" class="oe_inline"/>%
+                                <label for="margin" string="Margin on Rate" attrs="{'invisible': [ ('delivery_type', 'in', ('fixed', 'base_on_rule'))]}"/>
```

Comment:
remove when it's base on rule.
Use case: if the user want to set the carrier price on the rule, it doesn't need to compute the margin each time the carrier modify its price